### PR TITLE
macOS: Clean up create_ios_framework.py

### PR DESCRIPTION
**Pull request by cbracken**
Over time, this script and others in sky/tools have accumulated a lot of additional and sometimes duplicate code.

This is a first pass cleanup of create_macos_framework.py to extract common utility code to utils.py and refactor for better readability.

The iOS analogue of this patch was flutter/engine#54500.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cbracken** [comment]
This doesn't add/change any functionality; just makes use of the common code in `sky_utils` where possible. Not touching the framework creation and symlink wrangling in this one. 😉

File: sky/tools/create_macos_framework.py [hunk 1]
```diff
@@ -143,108 +130,78 @@ def regenerate_symlinks(fat_framework_bundle):
...
+    sky_utils.strip_binary(fat_framework_binary, unstripped_out)
+
+
+def zip_framework(dst):
+  framework_dst = os.path.join(dst, 'FlutterMacOS.framework')
+  sky_utils.write_codesign_config(os.path.join(framework_dst, 'entitlements.txt'), [])
+  sky_utils.write_codesign_config(
+      os.path.join(framework_dst, 'without_entitlements.txt'),
+      [
+          # TODO(xilaizhang): Remove the zip file from the path when outer zip is removed.
```
Comment (zanderso) [inline sky/tools/create_macos_framework.py:160]: nit: Xilai is probably not going to do this =(

**Review by zanderso — APPROVED** [review]

File: sky/tools/create_macos_framework.py (hunk 1 above)
Comment (cbracken) [reply sky/tools/create_macos_framework.py:160]: sgtm -- let me think of a name that isn't Xilai or me who also won't do this. 😂

Comment (cbracken) [reply sky/tools/create_macos_framework.py]: Alright, I have accepted the heavy burden of another TODO that I probably won't do.

File: sky/tools/create_macos_framework.py [hunk 2]
```diff
@@ -71,12 +62,7 @@ def main():
     print('Cannot find macOS x64 dylib at %s' % x64_dylib)
     return 1
 
-  if not os.path.isfile(DSYMUTIL):
-    print('Cannot find dsymutil at %s' % DSYMUTIL)
-    return 1
-
-  shutil.rmtree(fat_framework_bundle, True)
```
[the remark this answers is not in the document]
Comment (jmagman) [reply sky/tools/create_macos_framework.py:78]: @cbracken points out that `shutil.copytree` already calls `rmtree`.

**Comment by zanderso** [comment]
Reason for revert: Failing roll https://github.com/flutter/flutter/pull/153406

**Pull request merged**